workflow/state-machines.rst

```diff
@@ -190,8 +189,23 @@ Below is the configuration for the pull request state machine.
             ),
         ));
 
-You can now use this state machine by getting the ``state_machine.pull_request`` service::
+In a Symfony application using the
+:ref:`default services.yaml configuration <service-container-services-load-example>`,
+you can get this state machine by injecting the Workflow registry service::
+
+    // ...
+    use Symfony\Component\Workflow\Registry;
+
+    class SomeService
+    {
+        private $stateMachine;
+
+        public function __constructor(Registry $workflows)
+        {
+            $this->stateMachine = $workflows->get('pull_request');
```

Comment:
I would not do that inside the constructor. You'll never know when your service is initialized and if the registry is ready at this point which could lead to undesired side effects.